rust/src/reading.rs

```diff
@@ -299,6 +300,36 @@ impl Messages {
             constraints: None,
         }
     }
+
+    pub fn validate_constraint_type(&self) -> Result<(), String> {
+        for message in self.into_iter() {
+            match message.message_as_constraint_system() {
+                None => continue,
+                Some(cs) => {
+                    if cs.constraint_type() == ConstraintType::arithmetic {
+                        let constraints = cs.constraints().unwrap();
+                        for i in 0..constraints.len() {
+                            let constraint = constraints.get(i);
+                            let n_a = constraint.linear_combination_a().unwrap().variable_ids().unwrap().len();
+                            let n_b = constraint.linear_combination_b().unwrap().variable_ids().unwrap().len();
+                            let n_c = constraint.linear_combination_c().unwrap().variable_ids().unwrap().len();
+
+                            let is_pure_multiplication =
+                                n_a == 1 && n_b == 1 && n_c == 1;
+
+                            let is_pure_linear =
+                                n_a == 0 && n_b == 0;
+
+                            if !(is_pure_multiplication || is_pure_linear) {
```

Comment:
@tromer What do you think of this validation?

`n_a n_b n_c` are the number of terms/wires in the linear combinations `A*B=C`.

In an arithmetic circuit there are
* Multiplication gates with a single wire into each of A B C.
* And linear combinations C such that `C=0` (A and B are left empty).